### PR TITLE
Ports vampires. Or, bloodsuckers, rather.

### PR DESCRIPTION
**Pull request by MalricB**
I spent the past 8 hours on this give me a break I'll make a proper PR in a bit
This ports bloodsuckers from [this pr](https://github.com/Citadel-Station-13/Citadel-Station-13/pull/10011) made for Citadel.

Bloodsuckers:
-Technically considered crew and human, unlike changeling
-However, they are dead and fake to be living. (Potential issue depending on our silicon policy)
-Antag that ramps-up-over-time. This can be configured to make it slower don't panic at this.
-Abilities (can only unlock one every 15 minutes essentially):
 *Brawn = Snap out of cuffs or hit targets with terrible strength. (Damage depends on vampire rank, about a 15 second cooldown, about 30~ ish damage (?))
*Cloak = make yourself invisible in light darkness at the expense of walking (could be tweaked to make you invisible in light too)
*Feed = feed on a passively grabbed carbon stealthily (but can be interrupted if the target moves or you're moved, basically giving you out) OR on a aggressively grabbed carbon to knock them out and feed on them
*Fortitude = extra armor against almost everything, except burn iirc. at the expense of only being able to walk.
*Gohome = teleport to your lair when it's near daytime. Only given near daytime.
*Haste = lunge at a target turf, knocking or disarming carbons on the way. 15-20 second cooldown.
*Lunge = lunge into a target, aggressively grabbing them. They can still resist out of it or move out as they're still standing. However, if they're facing away from you or you're amidst the darkness, they'll be knocked down. (1v1 tool. 30 second cooldown.)
*Masquerade = make your heart beat once again, faking your undeath to medical scanners and the such.
*Trespass = teleport into a mist of smoke into the target location. (low cooldown but very low range, and very visible. not stealthy.)
*Veil = teleport into a mist of smoke to change appearances

Thrall abilities:
*Recuperate = use your tainted blood to heal.
Misc:
-They can make a lair by sleeping on a coffin, it will make the area their lair and as such it'll spawn rats, cobwebs, amongst other things.
Gamemode itself:
-Bloodsuckers have to avoid daytime. If they're caught during daytime, they'll burst into flames. One way to avoid daytime is to sleep in a coffin during it, which means they'll be literally in a deep-sleep state where nothing can wake them up unless daytime's over.
Objectives:
-Steal hearts. Can be cheesed by stealing monkey hearts.
-Survive.
-Make a lair.
-Enthrall someone and make sure they're in X department / or end up as a head of staff
TLDR:
Changelings but they ramp up over time instead and are not as yucky

Changelog
🆑
add: Adds Bloodsuckers, beware.
/🆑

### CODE REVIEW

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -124,9 +124,13 @@
 #define TRAIT_ALCOHOL_LIGHTWEIGHT	"alcohol_lightweight" //Skyrat port
 #define TRAIT_CURSED_BLOOD	"cursed_blood" //Yo dawg I heard you like bloodborne references so I put a
 #define TRAIT_RADRESONANCE "radioactive_resonance" //Hyperstation edit
-
 #define TRAIT_SWIMMING			"swimming"			//only applied by /datum/element/swimming, for checking
 #define TRAIT_CAPTAIN_METABOLISM "captain-metabolism"
+#define TRAIT_COLDBLOODED		"coldblooded"	// Your body is literal room temperature. Does not make you immune to the temp.
```
Comment (VladinXXV) [inline code/__DEFINES/traits.dm:129]: we already have this exact trait, the coldblooded one. this very file, line 176

Comment (MalricB) [reply code/__DEFINES/traits.dm:129]: whoops

Comment (MalricB) [reply code/__DEFINES/traits.dm]: qdeleted

**Comment by MalricB** [comment]
it compiles it works, tests went well, time to test live whenever
merge_my_pr

**Comment by MalricB** [comment]
stuff to change depending on opinions/how the rounds go:
-Thralls in general
-Ramp-up of bloodsucked (extend night time to 30 minutes instead of 15?)

File: code/__DEFINES/status_effects.dm [hunk 2]
```diff
@@ -45,6 +45,7 @@
 #define STATUS_EFFECT_SLEEPING /datum/status_effect/incapacitating/sleeping //the affected is asleep
 
 #define STATUS_EFFECT_TASED /datum/status_effect/electrode //the affected has been tased, preventing fine muscle control
```
Comment (ArchieBeepBoop) [inline code/__DEFINES/status_effects.dm:47]: This is a double define. Delete the previous one if we're going to use a new tase status effect.
(looking at https://github.com/Citadel-Station-13/Citadel-Station-13/pull/10011/files)

**Comment by MalricB** [comment]
okay update: it's functional but there's some bugs. particularly, bsuckers not getting their objectives in their notes (yet they still have it in antag panel ?????????)
and getting an extra hand on making sure they can't thrall more than 1 person

File: code/__DEFINES/status_effects.dm (hunk 2 above)
Comment (MalricB) [reply code/__DEFINES/status_effects.dm]: fixed, ty

**Comment by MalricB** [comment]
above changes applied.
many thanks to Azarak, Deltafire, Archie & JaySparrow

**Comment by Dahlular** [comment]
merging this because you wouldnt fucking shut up about it

**Comment by JohnFulpWillard** [comment]
massive win
